streaming/sender.c

```diff
@@ -70,8 +70,10 @@ static int rrdpush_sender_thread_custom_host_variables_callback(void *rrdvar_ptr
 }
 
 static void rrdpush_sender_thread_send_custom_host_variables(RRDHOST *host) {
+    sender_start(host->sender);
```

Comment:
Acquiring and releasing the lock twice (and also duplicating the copy from the build into the circular buffer) is less efficient. Was there a reason not to move the call into the locked section?

Reply:
Yes,  I moved it there and the problem persists.

Reply:
Are the callbacks deferred or do they happen during the call?